test/borda_clj/trace.clj

```diff
@@ -0,0 +1,43 @@
+(ns borda-clj.trace
+  (:require  [clojure.test :as t]
+             [borda-clj.client :as sut]
+             [clojure.core.async :as async :refer (<! <!! >! >!! alts! chan close! go go-loop onto-chan timeout)]))
+
+(defn run-trace [global-dimensions max-buffer-size trace]
+  (let [[report-ch measurements-ch kill-ch timeout-ch trace-ch] (repeatedly chan)
+        successful-send (fn [m] (>!! trace-ch {:send m}))
+        failed-send (fn [& args] (throw (RuntimeException. "boom")))
+        final-send (fn [m] (>!! trace-ch {:final-send m}))
+        send (fn [measurements] ((or (:trace-metadata (meta measurements))
+                                    final-send)
+                                measurements))
+        on-send-error (fn [m e]
+                        ;;(clojure.pprint/pprint e)
+                        (>!! trace-ch {:error {:measurements m}}))
+        flush-ch (sut/go-flush-measurements report-ch measurements-ch kill-ch global-dimensions ::bogus-interval send on-send-error (constantly timeout-ch))]
```

Comment:
core.async `go` and `go-loop` return a channel that yields the value of the last expression in the body.  We don't care about this value, but we do want to know when the `go-flush-measurements` go block is done.